vnext/Microsoft.ReactNative.IntegrationTests/JsiTurboModuleTests.cpp

```diff
@@ -0,0 +1,419 @@
+
+facebook::jsi::Object MySimpleTurboModule::getObject(facebook::jsi::Runtime &rt, const facebook::jsi::Object &arg) {
+  LogAction("getObject called", "OK");
+  return facebook::react::deepCopyJSIObject(rt, arg);
+}
+
+facebook::jsi::Object MySimpleTurboModule::getValue(
+    facebook::jsi::Runtime &rt,
+    double x,
+    const facebook::jsi::String &y,
+    const facebook::jsi::Object &z) {
+  LogAction("getValue called", "OK");
+  // Note: return type isn't type-safe.
+  facebook::jsi::Object result(rt);
+  result.setProperty(rt, "x", facebook::jsi::Value(x));
+  result.setProperty(rt, "y", facebook::jsi::String::createFromUtf8(rt, y.utf8(rt)));
+  result.setProperty(rt, "z", facebook::react::deepCopyJSIObject(rt, z));
+  return result;
+}
+
+void MySimpleTurboModule::getValueWithCallback(facebook::jsi::Runtime &rt, const facebook::jsi::Function &callback) {
+  LogAction("getValueWithCallback called", "OK");
+  callback.call(rt, facebook::jsi::String::createFromUtf8(rt, "value from callback!"));
+}
+
+facebook::jsi::Value MySimpleTurboModule::getValueWithPromise(facebook::jsi::Runtime &rt, bool error) {
+  LogAction("getValueWithPromise called", error);
+  return facebook::react::createPromiseAsJSIValue(
+      rt, [error](facebook::jsi::Runtime &rt2, std::shared_ptr<facebook::react::Promise> promise) {
+        if (error) {
+          promise->reject("intentional promise rejection");
+        } else {
+          promise->resolve(facebook::jsi::String::createFromUtf8(rt2, "result!"));
+        }
+      });
+}
+
+facebook::jsi::Object MySimpleTurboModule::getConstants(facebook::jsi::Runtime &rt) {
+  LogAction("getConstants called", "OK");
+  // Note: return type isn't type-safe.
+  facebook::jsi::Object result(rt);
+  result.setProperty(rt, "const1", facebook::jsi::Value(true));
+  result.setProperty(rt, "const2", facebook::jsi::Value(375));
+  result.setProperty(rt, "const3", facebook::jsi::String::createFromUtf8(rt, "something"));
+  return result;
+}
+
+struct MySimpleTurboModulePackageProvider
+    : winrt::implements<MySimpleTurboModulePackageProvider, IReactPackageProvider> {
+  void CreatePackage(IReactPackageBuilder const &packageBuilder) noexcept {
+    AddTurboModuleProvider<MySimpleTurboModule>(packageBuilder, L"MySimpleTurboModule");
+  }
+};
+
+} // namespace
+
+TEST_CLASS (JsiTurboModuleTests) {
+  TEST_METHOD(ExecuteSampleTurboModule) {
+    MySimpleTurboModule::s_actionIndex = -1;
+    TestAction expectedActions[] = {
+        TestAction{"voidFunc called", nullptr},
+        TestAction{"getBool called", true},
+        TestAction{"getBool result", true},
+        TestAction{"getBool called", false},
+        TestAction{"getBool result", false},
+        TestAction{"getNumber called", 5.0},
+        TestAction{"getNumber result", 5.0},
+        TestAction{"getNumber called", NAN},
+        TestAction{"getNumber result", NAN},
+        TestAction{"getNumber called", INFINITY},
+        TestAction{"getNumber result", INFINITY},
```

Comment:
nit: consider using the values in `std::numeric_limits` instead of the C `#defines`

Reply:
Good point! I will change it.

---
In reply to: [550078404](https://github.com/microsoft/react-native-windows/pull/6804#discussion_r550078404) [](ancestors = 550078404)